From c35afb2689ab051dec847236ced68dbc3a82bfee Mon Sep 17 00:00:00 2001
From: Igor Pecovnik <igor@armbian.com>
Date: Sat, 17 Oct 2026 16:57:53 +0200
Subject: [PATCH] board: bananapi: bpi-w2: turn on the eth1 jack LEDs

The NAT port LEDs sit on the nat_led_0 and nat_led_2 pads. Select their
LED function, as the BSP device tree does.

Signed-off-by: Igor Pecovnik <igor@armbian.com>
---
 board/bananapi/bpi-w2/bpi-w2.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/board/bananapi/bpi-w2/bpi-w2.c b/board/bananapi/bpi-w2/bpi-w2.c
index bb848409ef..36107bdfeb 100644
--- a/board/bananapi/bpi-w2/bpi-w2.c
+++ b/board/bananapi/bpi-w2/bpi-w2.c
@@ -4,8 +4,24 @@
  */
 
 #include <init.h>
+#include <asm/io.h>
+#include <linux/bitops.h>
+
+#define ISO_MUXPAD1		0x98007314
+#define  NAT_LED_0		GENMASK(17, 16)
+#define  NAT_LED_2		GENMASK(21, 20)
+#define  NAT_LED_FUNC		1
+
+/* The LEDs of the NAT port (eth1) jack, as the BSP device tree sets them */
+static void bpi_w2_nat_leds(void)
+{
+	clrsetbits_le32(ISO_MUXPAD1, NAT_LED_0 | NAT_LED_2,
+			NAT_LED_FUNC << 16 | NAT_LED_FUNC << 20);
+}
 
 int board_init(void)
 {
+	bpi_w2_nat_leds();
+
 	return 0;
 }
-- 
2.53.0

